Dev

boris-sulyagin opened the pull request.

Commit:
Спринт 12/25 → Тема 1/3: Shared Preferences → Урок 3/3

boris2034 committed:
Экран настроек, switch color

boris2034 committed:
Спринт 12

boris2034 committed:
Спринт 12 fix

IliaAnK commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -21,112 +22,181 @@ import androidx.recyclerview.widget.RecyclerView
 
-    private val tracksSearch = retrofit.create(ItunesSongSearch::class.java)
-    private val tracks = ArrayList<Track>()
+    private val serviceSearch = retrofit.create(ItunesSongSearch::class.java)
```
Рекомендация:
Создание всего ретрофит клиента можно вынести в отдельный класс/файл

boris2034 committed:
Спринт 12 fix

boris2034 committed:
Спринт 12 fix

Выполнение требований задания:
 ❌ Пользователь находится на экране «Поиск», поле поискового запроса пустое и находится в фокусе. При условии, что пользователь имеет непустую историю и не очищал историю поиска, он видит:

Заголовок «История поиска».
Список просмотренных ранее треков.
Кнопку «Очистить историю».
⚠️ Критические замечания:

Баг. Окружение: эмулятор Pixel 6, 33 Api, история переходов пуста
Кейс: перейти на экран "Поиск" -> нажать на поле ввода
ОР: после перехода на экран "Поиск" не отображаются элементы истории. Аналогично после нажатия на поле ввода
ФР: отображаются элементы истории;
Баг. Окружение: эмулятор Pixel 6, 33 Api, история переходов пуста
Кейс: перейти на экран "Поиск" -> выполнить поиск -> появился результат поиска -> стереть запрос с помощью клавиатуры или кнопки "Очистить"-> поле ввода пустое
ОР: последний результат поиска не отображается
ФР: отображается последний результат поиска;
Баг. Окружение: эмулятор Pixel 6, 33 Api, история переходов не пуста
Кейс: перейти на экран "Поиск" -> выполнить поиск -> появился результат поиска -> стереть запрос с помощью клавиатуры -> поле ввода пустое
ОР: поскольку поле ввода пустое и в фокусе отображается история
ФР: отображается последний результат поиска.

boris2034 committed:
Спринт 12 fix

 Критические замечания:
    - Баг. Окружение: эмулятор Pixel 6, 33 Api, история переходов не пуста
Кейс: перейти на экран "Поиск" -> выполнить поиск -> появился результат поиска -> стереть запрос с помощью клавиатуры -> отображается история -> очистить историю
ОР: поле ввода пустое, ничего не отображается
ФР: отображается последний результат поиска.
